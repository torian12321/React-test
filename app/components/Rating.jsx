var
React     = require('react'),
PropTypes = require('prop-types');

const Rating = props => {
    var
    highlights= (props.children > props.of) ? props.of : props.children,
    marks     = [];
    for (var i=0; i < props.of; i++) {
        if((i+1) > highlights){
            marks.push(<li className="o"></li>);
        }else{
            marks.push(<li></li>);
        }
    }

    return(
        <ul className={ 'rating ' + props.style }>
            { marks }
        </ul>
    );
};


Rating.propTypes = {
    children: PropTypes.number.isRequired,
    of      : PropTypes.number.isRequired,
    style   : PropTypes.string.isRequired
};
Rating.defaultProps = {
    children: 0,
    style   : '',       // info | success | danger | warning
    of      : 5
};

module.exports = Rating;
