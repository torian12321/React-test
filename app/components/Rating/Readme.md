Basic
```
<Rating />
```

Custom values
```
<div>
	<Rating val={3} of={10} />
	<Rating val={7} of={15} />
</div>
```