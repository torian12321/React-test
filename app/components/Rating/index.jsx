var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames');

const Rating = props => {
    let
    rate = (props.val > props.of) ? props.of : props.val,
    marks= [];

    for(let i=0, m=props.of; i < m; i++){
        marks.push(
            <li
                key      = {i}
                className= {(i+1) > rate ? 'o' : null}
            />
        );
    }

    return(
        <ul className={classNames('rating', props.className)}>
            {marks}
        </ul>
    );
};


Rating.propTypes = {
    /** If val>of, then val=of. */
    val      : PropTypes.number,
    of       : PropTypes.number,
    className: PropTypes.oneOf(['info', 'success', 'danger', 'warning'])
};
Rating.defaultProps = {
    val: 0,
    of : 5
};

module.exports = Rating;
