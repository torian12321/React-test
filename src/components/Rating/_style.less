@import '../../styles/conf/index.less';

.rating{
    .icon{
        fill  : @themeColor_1;
        margin: 0 .4em 0 0;
        &:last-child{
            margin: 0;
        }
        &.o{
            fill: @colorState_disable;
        }
    }
}