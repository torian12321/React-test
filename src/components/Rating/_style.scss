@import '../../styles/conf/index.scss';

// Vars
$rating-color-main: $themeColor_1 !default;

// Block
.rating {
  .icon {
    fill  : $rating-color-main;
    margin: 0 .4em 0 0;

    &:last-child {
      margin: 0;
    }
    &.o {
      fill: $colorState_disable;
    }
  }
}