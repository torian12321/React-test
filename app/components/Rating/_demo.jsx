import React from 'react'
import Demo, {props as P} from 'react-demo'
import Rating from './index'


module.exports = () => (
	<div>
		<Demo target={Rating} />
		<Demo 
			target={Rating}
			props ={{
				val: P.number(3),
				of : P.number(10)
			}}
		/>
	</div>
);