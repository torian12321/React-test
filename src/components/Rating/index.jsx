import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import Icon       from '../Icon';
import './_style.less';

class Rating extends React.PureComponent {
  render(){
    let
    rate  = this.props.val > this.props.of ? this.props.of : this.props.val,
    marks = [];

    for(let i=0, m=this.props.of; i < m; i++){
      marks.push(
        <Icon
          key      = {i}
          name     = {this.props.icon}
          color    = {(i+1) > rate ? null : this.props.color}
          className= {(i+1) > rate ? 'o'  : null}
        />
      );
    }

    return <div className={classNames('rating', this.props.className)}>{marks}</div>;
  }
};

Rating.propTypes = {
  /** If val>of, then val=of. */
  val  : PropTypes.number,
  of   : PropTypes.number,
  color: PropTypes.string,
  icon : PropTypes.oneOf(['star', 'chat', 'cog', 'heart', 'dollar'])
};
Rating.defaultProps = {
  val : 0,
  of  : 5,
  icon: 'star'
};

module.exports = Rating;
