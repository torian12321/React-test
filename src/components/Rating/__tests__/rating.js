import React    from 'react';
import renderer from 'react-test-renderer';
import Rating   from '../index';

describe('component', () => {
  describe('<Rating />', () => {
    
    it('should render correctly', () => {
      const comp = renderer.create(
        <Rating />
      ).toJSON();

      expect(comp).toMatchSnapshot();
      expect(comp.children).toHaveLength(5);
      expect(comp.props.className).toEqual('rating');
    });

    it('should render 7 starts', () => {
      const comp = renderer.create(
        <Rating val={5} of={7} />
      ).toJSON();

      expect(comp).toMatchSnapshot();
      expect(comp.children).toHaveLength(7);
    });

    it('should render hearts', () => {
      const comp = renderer.create(
        <Rating icon='heart' />
      ).toJSON();

      expect(comp).toMatchSnapshot();
    });

  });
});
