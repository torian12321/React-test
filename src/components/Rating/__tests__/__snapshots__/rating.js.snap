// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Rating /> should render 7 starts 1`] = `
<div
  className="rating"
>
  <svg
    className="icon"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
</div>
`;

exports[`component <Rating /> should render correctly 1`] = `
<div
  className="rating"
>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M16 0l5 11 11 1-9 7 3 12-10-6-10 6 3-12-9-7 11-1"
      />
    </g>
  </svg>
</div>
`;

exports[`component <Rating /> should render hearts 1`] = `
<div
  className="rating"
>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 10c0-4 3-8 8-8 4 0 7 3 8 4 1-1 4-4 8-4 6 0 8 4 8 8 0 8-14 19-16 20-2-1-16-12-16-20"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 10c0-4 3-8 8-8 4 0 7 3 8 4 1-1 4-4 8-4 6 0 8 4 8 8 0 8-14 19-16 20-2-1-16-12-16-20"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 10c0-4 3-8 8-8 4 0 7 3 8 4 1-1 4-4 8-4 6 0 8 4 8 8 0 8-14 19-16 20-2-1-16-12-16-20"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 10c0-4 3-8 8-8 4 0 7 3 8 4 1-1 4-4 8-4 6 0 8 4 8 8 0 8-14 19-16 20-2-1-16-12-16-20"
      />
    </g>
  </svg>
  <svg
    className="icon o"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 10c0-4 3-8 8-8 4 0 7 3 8 4 1-1 4-4 8-4 6 0 8 4 8 8 0 8-14 19-16 20-2-1-16-12-16-20"
      />
    </g>
  </svg>
</div>
`;
